### PR TITLE
Pass test JVM arguments to test nodes

### PR DESCRIPTION
**Pull request by jasontedor**
This pull request modifies the integration test node spinup to pass
through the system property `tests.jvm.argline`. The main purpose of
passing through this `tests.jvm.argline` is so that we can continue to
test with G1GC on CI. Thus, also added is a test that if G1GC is
enabled, then `JvmInfo#useG1GC` returns the right thing on HotSpot.

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/cluster/routing/allocation/ShardStateIT.java [hunk 1]
```diff
@@ -48,8 +48,10 @@ public void testPrimaryFailureIncreasesTerm() throws Exception {
         indicesService.indexService(resolveIndex("test")).getShard(shard).failShard("simulated test failure", null);
 
         logger.info("--> waiting for a yellow index");
-        assertBusy(() -> assertThat(client().admin().cluster().prepareHealth().get().getStatus(),
-            equalTo(ClusterHealthStatus.YELLOW)));
+        // JDK 9 type inference gets confused, so we have to help the
+        // type inference
+        assertBusy((Runnable)(() -> assertThat(client().admin().cluster().prepareHealth().get().getStatus(),
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/cluster/routing/allocation/ShardStateIT.java]: I usually find it cleaner to make a variable of type Runnable in cases like this. Doesn't matter though.

Comment (jasontedor) [inline core/src/test/java/org/elasticsearch/cluster/routing/allocation/ShardStateIT.java]: Sure, I'll keep that in mind in the future. :smile:

File: core/src/test/java/org/elasticsearch/monitor/jvm/JvmInfoTests.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        final boolean positiveUseG1GCIsLastUseG1GC = arglineContainsG1GC && argline.charAt(index - 1) == '+';
+        // for JDK 9 the default collector when no collector is specified is G1 GC
+        final boolean versionIsAtLeastJava9 = JavaVersion.current().compareTo(JavaVersion.parse("9")) >= 0;
+        final boolean noOtherCollectorSpecified =
+                argline == null ||
+                        (!argline.contains("-XX:+UseParallelGC") &&
+                                !argline.contains("-XX:+UseParallelOldGC") &&
+                                !argline.contains("-XX:+UseSerialGC") &&
+                                !argline.contains("-XX:+UseConcMarkSweepGC"));
+        return positiveUseG1GCIsLastUseG1GC || (versionIsAtLeastJava9 && noOtherCollectorSpecified);
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/monitor/jvm/JvmInfoTests.java]: +++

File: core/src/test/java/org/elasticsearch/monitor/jvm/JvmInfoTests.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+public class JvmInfoTests extends ESTestCase {
+
+    public void testUseG1GC() {
+        // if we are running on HotSpot, and the test JVM was started
+        // with UseG1GC, then JvmInfo should successfully report that
+        // G1GC is enabled
+        if (Constants.JVM_NAME.contains("HotSpot")) {
+            assertEquals(Boolean.toString(isG1GCEnabled()), JvmInfo.jvmInfo().useG1GC());
+        } else {
+            assertEquals("unknown", JvmInfo.jvmInfo().useG1GC());
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/monitor/jvm/JvmInfoTests.java]: I don't suspect we'll test this branch frequently. Which is fine.

**Comment by nik9000** [comment]
LGTM

File: core/src/test/java/org/elasticsearch/monitor/jvm/JvmInfoTests.java (hunk 3 above)
Comment (jasontedor) [inline core/src/test/java/org/elasticsearch/monitor/jvm/JvmInfoTests.java]: I also suspect that *we* will not be testing this branch, but I wanted to be correct. :smile:

**Comment by jasontedor** [comment]
Thanks @nik9000!

**Pull request merged**